Source/Scene/SceneFramebuffer.js

```diff
@@ -33,138 +24,56 @@ function SceneFramebuffer() {
-  post._depthStencilIdTexture =
-    post._depthStencilIdTexture && post._depthStencilIdTexture.destroy();
-  post._depthStencilIdRenderbuffer =
-    post._depthStencilIdRenderbuffer &&
-    post._depthStencilIdRenderbuffer.destroy();
-
-  post._framebuffer = undefined;
-  post._idFramebuffer = undefined;
-  post._colorTexture = undefined;
-  post._idTexture = undefined;
-  post._depthStencilTexture = undefined;
-  post._depthStencilRenderbuffer = undefined;
-  post._depthStencilIdTexture = undefined;
-  post._depthStencilIdRenderbuffer = undefined;
+  post._colorFramebuffer.destroyResources();
+  post._idFramebuffer.destroyResources();
 }
 
+Object.defineProperties(SceneFramebuffer.prototype, {
+  framebuffer: {
+    get: function () {
+      return this._colorFramebuffer.framebuffer;
+    },
+  },
+  idFramebuffer: {
+    get: function () {
+      return this._idFramebuffer.framebuffer;
+    },
+  },
+});
+
 SceneFramebuffer.prototype.update = function (context, viewport, hdr) {
   var width = viewport.width;
   var height = viewport.height;
-  var colorTexture = this._colorTexture;
-  if (
-    defined(colorTexture) &&
-    colorTexture.width === width &&
-    colorTexture.height === height &&
-    hdr === this._useHdr
-  ) {
-    return;
-  }
-
-  destroyResources(this);
-  this._useHdr = hdr;
-
-  var pixelDatatype = hdr
-    ? context.halfFloatingPointTexture
-      ? PixelDatatype.HALF_FLOAT
-      : PixelDatatype.FLOAT
-    : PixelDatatype.UNSIGNED_BYTE;
-  this._colorTexture = new Texture({
-    context: context,
-    width: width,
-    height: height,
-    pixelFormat: PixelFormat.RGBA,
-    pixelDatatype: pixelDatatype,
-    sampler: Sampler.NEAREST,
-  });
-
-  this._idTexture = new Texture({
-    context: context,
-    width: width,
-    height: height,
-    pixelFormat: PixelFormat.RGBA,
-    pixelDatatype: PixelDatatype.UNSIGNED_BYTE,
-    sampler: Sampler.NEAREST,
-  });
-
-  if (context.depthTexture) {
-    this._depthStencilTexture = new Texture({
-      context: context,
-      width: width,
-      height: height,
-      pixelFormat: PixelFormat.DEPTH_STENCIL,
-      pixelDatatype: PixelDatatype.UNSIGNED_INT_24_8,
-      sampler: Sampler.NEAREST,
-    });
-    this._depthStencilIdTexture = new Texture({
-      context: context,
-      width: width,
-      height: height,
-      pixelFormat: PixelFormat.DEPTH_STENCIL,
-      pixelDatatype: PixelDatatype.UNSIGNED_INT_24_8,
-      sampler: Sampler.NEAREST,
-    });
-  } else {
-    this._depthStencilRenderbuffer = new Renderbuffer({
-      context: context,
-      width: width,
-      height: height,
-      format: RenderbufferFormat.DEPTH_STENCIL,
-    });
-    this._depthStencilIdRenderbuffer = new Renderbuffer({
-      context: context,
-      width: width,
-      height: height,
-      format: RenderbufferFormat.DEPTH_STENCIL,
-    });
+  var depthTexture = context.depthTexture;
+  if (this._colorFramebuffer.isDirty(width, height, hdr)) {
```

Comment:
Another `isDirty` check that can be removed if either all or none of the framebuffers will be dirty during an update.